ssh_automated_tests/core/shell.py

```diff
@@ -0,0 +1,63 @@
+from abc import ABC, abstractmethod
+from time import sleep
+
+from ssh_automated_tests.core.connection import Connection
+
+
+class Shell(ABC):
+    @abstractmethod
+    def execute(self, command: str):
+        pass
+
+    @abstractmethod
+    def execute_with_output(self, command: str):
+        pass
+
+    @abstractmethod
+    def close(self):
+        pass
+
+
+class InteractiveShell(Shell):
```

Comment:
Why we have to declare all methods in interface? As i thought interface - it's generalized thing and our `InteractiveShell` class implements it and adds its own functionality. Here can be another implementation like `SimpleShell` which don't need such methods as `append_file_with_vim` or `send_key`.